openedx/openedx-platform
# MICROBA-1011 | Remove duplicate allowlist entries and add constraint preventing future occurrences

## Description

[MICROBA-1011]
- Add data migration to remove the duplicate certificate allowlist entries before adding the new constraint to the CertificateWhitelist model.
- Add `unique_together` constraint to the CertificateWhitelist model. A learner should only have a single allowlist entry per course-run.

## Testing instructions

- Manually insert multiple entries (for multiple users) to the `CERTIFICATES_CERTIFICATEWHITELIST` table in the LMS:
`insert into certificates_certificatewhitelist (course_id, whitelist, created, user_id) values ('<course_run_key>', 1, NOW(), <user_id>);`
- Take note of the entries in the table afterwards.
- Run migrations
- Verify that there is only one entry per user/course-run combo. The `created` date should match the _first_/"original" entry per user/course-run combo.


[MICROBA-1011]: https://openedx.atlassian.net/browse/MICROBA-1011
